addons/project/models/project_task.py

```diff
@@ -1072,33 +1018,42 @@ def _load_records_create(self, vals_list):
 
     @api.model_create_multi
     def create(self, vals_list):
+        # Some values are computed and must be written as sudo for portal users,
+        # because they do not have access to these fields.
+        computed_vals_list = [{} for _ in vals_list]
```

Comment:
That part is confusing to me.  The "computed" fields you mention are not "computed" in the ORM sense.  So please use another word than "computed".  Those are values determined by `create` itself.

Also, please write a comment that explains the problem  Some field values are automatically determined by this override of `create()`.  However, some of those fields may not be written by portal users.  In order to work around the issue, we avoid to `sudo()` the call to `super().create()` because of the risks with x2many fields.  Instead, we call `super().create()` the normal way, and `sudo().write()` the extra field values that may not be written by the user.